src/charm.py

```diff
@@ -79,7 +81,9 @@ def __init__(self, *args):
             return
 
         self.topology = JujuTopology.from_charm(self)
-        self.otel_config = Config.default_config()
+        self.otel_config = Config(
+            cast(bool, self.model.config.get("tls_insecure_skip_verify"))
+        ).default_config()
```

Comment:
From a builder perspective, this should be toggleable throughout the building process. Currently, we get only one chance to set it - at init time.
A better API would be:
```python
config.insecure_skip_verify(True)
config.insecure_skip_verify(False)
rendered = config.yaml
```